### PR TITLE
Add default configuration tests

### PR DESCRIPTION
**Pull request by StepanVovkPLV**
Added tests : 
```
test_vlan_all_supported_modes_broadcast
test_vlan_all_supported_modes_multicast
test_vlan_all_supported_modes_unicast
```
Depends on [PR](https://github.com/PLVision/dent-testing/pull/16/files)

### CODE REVIEW

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/utils/test_utils/br_utils.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+       device: Dent device host_name
+       dut_ports (list): List of ports of the dent device.
+       port_vlan_map (tuple): Tuple of dicts. Ex:
+            port_vlan_map = (
+                {"port": 0, "settings": [{"vlan": 2, "untagged": False, "pvid": False},)
+    """
+    for config in port_vlan_map:
+        port = dut_ports[config["port"]]
+        out = await BridgeVlan.add(input_data=[{device: [
+            {"device": port, "vid": setting["vlan"], "pvid": setting["pvid"], "untagged": setting["untagged"], } for
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/utils/test_utils/br_utils.py]: please move `for` to new line

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[3]][0]["name"], dev_groups[tg_ports[1]][0]["name"],
+                dev_groups[tg_ports[2]][0]["name"]]
+
+    # 4. Setup packet stream(s).
+    streams = {f"traffic with VLAN ID: {vlan}": {
+        "type": "raw",
+        "protocol": "802.1Q",
+        "ip_source": tx_ports,
+        "ip_destination": rx_ports, "srcMac": "aa:bb:cc:dd:ee:ff",
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: move `srcMac` to new line

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    # 5.1 Verify traffic.
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    ti_to_rx_port_map = get_traffic_port_vlan_mapping(streams, port_map, tg_ports)
+    for row in stats.Rows:
+        if row["Rx Port"] in ti_to_rx_port_map[row["Traffic Item"]]:
+            assert tgen_utils_get_loss(row) == 0.000, \
+                f'Traffic leak for traffic item: {row["Traffic Item"]} on port {row["Rx Port"]}'
+        else:
+            assert tgen_utils_get_loss(row) == 100.000, \
+                f'No traffic for traffic item : {row["Traffic Item"]} on port {row["Rx Port"]}'
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: `Traffic for traffic item ...` or `Unexpected traffic ...` or `Expected 100% loss ...`

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[x]][0]["name"] for x in range(1, 4)]
+
+    mac = {tg_ports[idx]: f"02:00:00:00:00:0{idx + 1}" for idx in range(4)}
+
+    vlan_to_mac = {2: "02:00:00:00:00:02",
+                   3: "02:00:00:00:00:03",
+                   4: "02:00:00:00:00:04",
+                   "X": "02:00:00:00:00:04",
+                   1: "02:00:00:00:00:04", }
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py]: You already have mac addrs in `mac` dict:
```python
vlan_to_mac = {
    2: mac[tg_ports[1]],
    3: mac[tg_ports[2]],
    4: mac[tg_ports[3]],
    "X": mac[tg_ports[3]],
    1: mac[tg_ports[3]],
}
```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+        "vlanID": vlan, } for vlan in packet_vids if vlan != "X"}
+
+    streams.update({
+        f"{tg_ports[0]} -> X": {
+            "type": "raw",
+            "protocol": "802.1Q",
+            "ip_source": tx_ports,
+            "ip_destination": rx_ports,
+            "srcMac": mac[tg_ports[0]],
+            "dstMac": mac[tg_ports[3]], }})
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py]: [when to use trailing commas: https://peps.python.org/pep-0008/](https://peps.python.org/pep-0008/#when-to-use-trailing-commas)

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py (hunk 4 above)
Comment (StepanVovkPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py]: i believe this needed to distinguish from which port to send packet with certain vlan

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+    tg_ports = tgen_dev.links_dict[device][0]
+    dut_ports = tgen_dev.links_dict[device][1]
+
+    await configure_bridge_setup(device, dut_ports)
+
+    # 2. Set links to vlans.
+    await configure_vlan_setup(device, port_map, dut_ports)
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: Once you do VLAN configuration it is not a default one anymore.

Comment (anton7811) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: Your port_map corresponds to a default configuration, but you should not do any VLAN configuration steps. Just add ports to a VLAN-aware bridge.

**Review by anton7811 — CHANGES_REQUESTED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py (hunk 6 above)
Comment (StepanVovkPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: im adding `default_pvid=1` on 49 line, kind of the same?

Comment (SerhiyBoikoPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: Your test is named `test_vlan_*default_configuration*_broadcast`. You don't need to call `configure_vlan_setup` here.

**Review by SerhiyBoikoPLV — CHANGES_REQUESTED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py (hunk 6 above)
Comment (StepanVovkPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: I'm adding `VLAN 2` as well, that's why calling `configure_vlan_setup `

Comment (SerhiyBoikoPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: Then I don't understand this test setup. That wouldn't make it `default_configuration` anymore.

Comment (StepanVovkPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: This is how MRVL did they setup with commands ...

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+        "ip_destination": rx_ports,
+        "src_mac": src_mac,
+        "dst_mac": dst_mac
+        }})
+
+    streams.update({f'{tg_ports[3]} -> {tg_ports[0]}': {
+        "type": "raw",
+        "protocol": "802.1Q",
+        "ip_source": dev_groups[tg_ports[0]][0]["name"],
+        "ip_destination": dev_groups[tg_ports[3]][0]["name"],
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: your stream name and ip_source/ip_destination do not match

**Review by SerhiyBoikoPLV — APPROVED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+              pytest.mark.usefixtures("cleanup_bridges", "cleanup_tgen")]
+
+packet_vids = ['X', 1, 2]  # VLAN tag number contained in the transmitted packet
+port_map = ({"port": 0, "settings": [{"vlan": 1, "untagged": True, "pvid": True}]},
+            {"port": 1, "settings": [{"vlan": 1, "untagged": True, "pvid": True}]},
+            {"port": 2, "settings": [{"vlan": 1, "untagged": True, "pvid": True}]},
+            {"port": 3, "settings": [{"vlan": 1, "untagged": True, "pvid": True}]})
+
+
+async def test_vlan_default_configuration_with_broadcast(testbed):
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py]: Looks like we can parametrize those tests.

**Review by anton7811 — CHANGES_REQUESTED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+            "ip_destination": dev_groups[tg_ports[0]][0]["name"],
+            "src_mac": dst_mac,
+            "dst_mac": src_mac,
+            "vlanID": 1
+        }})
+
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    # 4. Send traffic to rx_ports.
+    if traffic_type == "unicast":
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py:172]: Please clarify that you do this pre-send to learn frames. Do the same for other tests.

**Comment by StepanVovkPLV** [comment]
Closed in favor of https://github.com/PLVision/dent-testing/pull/42

**Pull request closed without merging**